### ISSUE TITLE
Enforce length check style in the `explicit-length-check` rule

### ISSUE THREAD

**Issue by jfmengels** [sindresorhus/eslint-plugin-unicorn#56]
`explicit-length-check` checks that `length` properties get compared to a value instead of checking the truthiness of the property.

In https://github.com/avajs/eslint-plugin-ava/pull/149#pullrequestreview-3799305, we remembered that there are some opinions on how that value then gets compared :sweat_smile: 

For instance:

``` js
a.length === 0
// vs
a.length < 1 // not sure I've ever seen this one TBH

a.length !== 0
// vs
a.length > 0
// vs
a.length >= 1

// + some I might have forgotten
```

It could be nice to be able to specify how you want your comparisons to be made, and make that consistent across your project.

I'm not sure what the options would look like. At the top of my head, I'm thinking:

``` json
{
  "unicorn/explicit-length-check": ["error", {
    "empty": ["eq", 0] | ["lt", 1], // one or the other
    "not-empty": ["ne", 0] | ["gt", 0] | ["gte", 1]
  }]
}
```

Should one be omitted (`empty` for instance), then you'd simply get no errors reports on how that one gets checked (both `a.length < 1` `a.length === 0` would be fine).

This sounds pretty complicated at the moment (especially if we want to support `==` too), so I'm not too fond of it at the moment, especially it would not make sense to have `empty: ["equal", 1200]` as the config. Maybe we could simplify it to

``` json
{
  "unicorn/explicit-length-check": ["error", {
    "empty": "eq" | "lt",
    "not-empty": "ne" | "gt" | "gte"
  }]
}
```

and then we have predefined values associated to each of those operators. We could also have longer names for the operators, but you'd then get `greater-than-or-equal` (which is okay, but pretty long).

I'm kind of proposing this system so that we could have the same system for a different rule checking the `indexOf` uses. It might be nice to think of a similar "API" for it.

Oh, and obviously, this should not report any check for specific indices: `a.length === 2` and `a.length == 2` should always be fine, whatever the settings may be.

**Comment by sindresorhus** [comment]
This sounds very complicated. I wonder if we could just enforce `a.length > 0` and wait adding options until someone complains?

**Comment by jfmengels** [comment]
Yes, it's way too complicated for what it is. I'm fine with just enforcing `a.length > 0`, and I'll cross my fingers hoping for nobody to complain.

**Comment by pvskarthikeya** [comment]
@sindresorhus @jfmengels I have started working on this. Looks a little complicated, will post if I need help. Thank you.
